public/main.js

```diff
@@ -0,0 +1,261 @@
+       block.querySelector('.product-price').textContent = `₽ ${product.quantity*product.price}`;
+    }
+    _init(){
+        document.querySelector('.cart-button').addEventListener('click', () => {
+            document.querySelector(this.container).classList.toggle('invisible');
+        });
+        document.querySelector(this.container).addEventListener('click', e => {
+           if(e.target.classList.contains('mines')){
+               this.removeProduct(e.target);
+           }
+        })
+        document.querySelector(this.container).addEventListener('click', e => {
+          if(e.target.classList.contains('plus')){
+              this.addProduct(e.target);
+          }
+      })
+    }
+}
+
+class CartItem extends Item{
+    constructor(el){
+        super(el);
+        this.quantity = el.quantity;
+    }
+    render(){
+    return `<div class="cart-item" data-id="${this.id_product}">
+            <div class="product-bio">
+            <h3 class="product-title">${this.product_name}</h3>
+            <button class="plus" data-id="${this.id_product}">+</button>
+            <span class="product-quantity">${this.quantity}</span>
+            <button class="mines" data-id="${this.id_product}">&minus;</button>
+        <p class="product-single-price">${this.price} ₽</p>
+        </div>
+        <div class="right-block">
+            <p class="product-price">₽ ${this.quantity*this.price}</p>
+        </div>
+        </div>`
+    }
+}
+const list2 = {
+    ProductsList: ProductItem,
+    Cart: CartItem
+};
+
+class ContactForm {
+    constructor(container = '.cart'){
+        this.container = container; 
+        this.userName = [];
+        this.userPhone = [];
+        this.userEmail = [];
+        this.render();
+        this._init();  
+    }
+
+    render(){
+        return new Promise((res, req) => {
+            const container = document.createElement('form');
+            container.classList.add('form-box');
+            container.classList.add('_margin');
+            container.innerHTML = `
+                        <input class ="userName" type="text" placeholder="Имя/Name">
+                        <input class ="userPhone " type="tel" placeholder="+7(000)000-0000">
+                        <input class ="userEmail " type="email" placeholder="mymail@mail.ru">
+                        <button type="submit" class ="cart-button submit">Отправить</button>
+                        <p class ="form-text_error"></p>`
+            document.querySelector('.cart').appendChild(container);
+        })
+    }
+
+    validName(value) {
+        let rex = /^[A-Za-zА-Яа-я ]+$/;
+        this.userName = value.match(rex);
+        console.log(this.userName);
+    }
+    validPhone(value) {
+        let rex = /^\+\d{1}\(\d{3}\)\d{3}-\d{4}$/;
+        this.userPhone = value.match(rex);
+        console.log(this.userPhone);
+    }
+    validEmail(value) {
+        let rex = /^[-._a-z0-9]+@(?:[a-z0-9][-a-z0-9]+\.)+[a-z]{2,6}$/;
```

Comment:
Сможете доработать, чтобы email адреса могли начинаться только на букву?
валидные email:
a@test.ru
a.a@test.ru
a123@test.ru

Невалидные:
1@test.ru
1a@test.ru
a.......a@test.ru

Reply:
```suggestion
        let rex = /^[a-z]+([.]{1})?[-_a-z0-9]+@(?:[a-z0-9][-a-z0-9]+\.)+[a-z]{2,6}$/;
```
Изменил содержимое на такой вариант. Сложновато было с точками